### PR TITLE
feat(chores): Add missing tools to nix-shell

### PR DESCRIPTION
**Pull request by mhkarimi1383**
## Pull Request template

Please, go through these steps before you submit a PR.

**Why is this PR required? What issue does it fix?**: controller-tools and some other tools are missing from `nix-shell`

**What this PR does?**: adds corresponding packages

**Does this PR require any upgrade changes?**: No

**If the changes in this PR are manually verified, list down the scenarios covered:**:

just run `nix-shell`

**Any additional information for your reviewer?** :
_Mention if this PR is part of any design or a continuation of previous PRs_

**Checklist:**
- [ ] Fixes #<issue number>
- [x] PR Title follows the convention of  `<type>(<scope>): <subject>`
- [ ] Has the change log section been updated?
- [ ] Commit has unit tests
- [ ] Commit has integration tests
- [x] (Optional) Are upgrade changes included in this PR? If not, mention the issue/PR to track:
- [x] (Optional) If documentation changes are required, which issue on https://github.com/openebs/openebs-docs is used to track them:

### CODE REVIEW

**Comment by mhkarimi1383** [comment]
I also want to add .envrc to automatically load that

**Comment by tiagolobocastro** [comment]
Pardon my ignorance, what's in `kubernetes-controller-tools` ?

**Comment by mhkarimi1383** [comment]
@tiagolobocastro 
controller-gen cli

**Review by tiagolobocastro — APPROVED** [review]

**Comment by dsharma-dc** [comment]
Though nothing harmful, do we really need this in product nix shell ? The required controller-gen is currently installed from the Makefile in the repo and is used to generate manifests.

**Comment by mhkarimi1383** [comment]
In my NixOS I ran `make -j 1` and controller-gen not found and with the latest one from nix repos (unstable) I got unrecognized option error for controller-gen command

**Comment by tiagolobocastro** [comment]
> Though nothing harmful, do we really need this in product nix shell ? The required controller-gen is currently installed from the Makefile in the repo and is used to generate manifests.

Downloaded binaries are not likely to work on nix and NixOS, may require patching et all.
It's simpler to just put it on the nix-shell rather than figure out why it doesn't work.

**Review by dsharma-dc — APPROVED** [review]

**Comment by mhkarimi1383** [comment]
I made this PR draft since I see that some more tools are missing....

I want to add every required tools to shell.nix to make easier to work for NixOS users :)

**Comment by mhkarimi1383** [comment]
Commit notes/descriptions are important

**Comment by mhkarimi1383** [comment]
Also we can change all of the actions to use nix shell, so changing go version, etc., will be easier

**Comment by tiagolobocastro** [comment]
Sounds good, just let us know when this is working and ready to merge 👍

**Comment by mhkarimi1383** [comment]
@tiagolobocastro 
Looks like ready, But since I'm getting https://github.com/openebs/lvm-localpv/issues/353 error I can't be sure about that

File: coverage.txt
Comment (dsharma-dc) [inline coverage.txt:1]: If I'm not mistaken, this is an autogenerated file and doesn't need to be checked-in.

Comment (mhkarimi1383) [reply coverage.txt:1]: Yes, It should be in `.gitignore` :)
I have pushed that file by mistake

**Comment by mhkarimi1383** [comment]
Almost complete,
I have isolated nix-shell's gopath and tempdir, Since installing specific version of some packages is a bit messy

File: shell.nix [hunk 1]
```diff
@@ -5,8 +5,24 @@ in
...
   ];
-}
\ No newline at end of file
+  shellHook = ''
+    export GOPATH=$(pwd)/nix/.go
+    export GOCACHE=$(pwd)/nix/.go/cache
+    export TMPDIR=$(pwd)/nix/.tmp
+    export PATH=$GOPATH/bin:$PATH
+    mkdir -p "$TMPDIR"
+    go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.4.0
```
Comment (tiagolobocastro) [inline shell.nix]: I think this is now the same the Makefile is installing? Do we need to do it here?

Comment (mhkarimi1383) [reply shell.nix]: But setting env variables was required, so I can run `make bootstrap` instead of `go install`

Comment (tiagolobocastro) [reply shell.nix]: Yeah that's fine, just wondering about specifically the install to avoid having to modify this file once we update the Makefile

Comment (tiagolobocastro) [reply shell.nix]: That said we also have the specific go version here, so we might need to change anyway :)

Comment (mhkarimi1383) [reply shell.nix]: Yeah, Also if we upgrade nixpkgs specified go version will not be available

Comment (mhkarimi1383) [reply shell.nix]: I would recommend changing the entire CI system to use nix-shell just like lint action

With that we will have a single source of truth :)

Comment (mhkarimi1383) [reply shell.nix]: Switched to `make bootstrap`, but getting

```
go: github.com/axw/gocov/gocov@latest (in github.com/axw/gocov@v1.2.1): go.mod:3: invalid go version '1.22.0': must match format 1.23
go.mod:5: unknown directive: toolchain
```

in the logs

but tools are installed correctly as far as I can tell

Comment (mhkarimi1383) [reply shell.nix]: Also getting

```
  go: gopkg.in/matm/v1/gocov-html@latest: unrecognized import path "gopkg.in": parse https://gopkg.in/?go-get=1: no go-import meta tags ()
```


Fixed in https://github.com/openebs/lvm-localpv/pull/354

File: shell.nix [hunk 2]
```diff
@@ -5,8 +5,25 @@ in
...
+    gnumake
+    minikube
     semver-tool
     yq-go
-    chart-testing
   ];
-}
\ No newline at end of file
+  shellHook = ''
+    export HOME=${builtins.getEnv "HOME"}
```
Comment (tiagolobocastro) [inline shell.nix:21]: why is this needed?

Comment (mhkarimi1383) [reply shell.nix:21]: In GitHub actions it's required to set HOME specifically, it will fail if you don't set that (in both action file and shell.nix)

Comment (mhkarimi1383) [reply shell.nix:21]: You can checkout previous commits action runs

**Review by dsharma-dc — APPROVED** [review]

File: shell.nix (hunk 2 above)
Comment (tiagolobocastro) [reply shell.nix:21]: Interesting, was not aware of this :+1:

**Review by tiagolobocastro — APPROVED** [review]
